docs/examples/files.md

```diff
@@ -0,0 +1,231 @@
+`pydantic` is a great tool for validating data coming from various sources.
+In this section, we will look at how to validate data from different types of files.
+
+!!! Note:
+    If you're using any of the below file formats to parse configuration / settings, you might want to
+    consider using the [`pydantic-settings`][pydantic_settings] library, which offers builtin
+    support for parsing this type of data.
+
+## JSON data
+
+`.json` files are a common way to store key / value data in a human-readable format.
+Here is an example of a `.json` file:
+
+```json
+{
+    "name": "John Doe",
+    "age": 30,
+    "email": "john@example.com"
+}
+```
+
+To validate this data, we can use a `pydantic` model:
+
+```python test="skip"
+from pydantic import BaseModel, EmailStr, PositiveInt
+
+
+class Person(BaseModel):
+    name: str
+    age: PositiveInt
+    email: EmailStr
+
+
+with open('person.json') as f:
+    json_string = f.read()
+    person = Person.model_validate_json(json_string)
+    print(repr(person))
+    #> Person(name='John Doe', age=30, email='john@example.com')
+```
+
+If the data in the file is not valid, `pydantic` will raise a [`ValidationError`][pydantic_core.ValidationError].
+Let's say we have the following `.json` file:
+
+```json
+{
+    "age": -30,
+    "email": "not-an-email-address"
+}
+```
+
+This data is flawed for three reasons:
+1. It's missing the `name` field.
+2. The `age` field is negative.
+3. The `email` field is not a valid email address.
+
+When we try to validate this data, `pydantic` raises a [`ValidationError`][pydantic_core.ValidationError] with all of the
+above issues:
+
+```python test="skip"
+from pydantic import BaseModel, EmailStr, PositiveInt, ValidationError
+
+
+class Person(BaseModel):
+    name: str
+    age: PositiveInt
+    email: EmailStr
+
+
+with open('person.json') as f:
+    json_string = f.read()
```

Comment:
same note here about using pathlib to read the file rather than `open` and `f.read`